### DISCUSSION TITLE
Weird actions merging with REFINE in RADB

### DISCUSSION THREAD

**Discussion by SichangHe** [SichangHe/internet_route_verification#128] [General]
https://github.com/SichangHe/internet_route_verification/commit/8db33f6d7490b854f0a5a276ccd12a1f58b078d2 and https://github.com/SichangHe/internet_route_verification/commit/f0a79d1f1bc67f7acb87cb4c1263bb4bb33cc315, and my log showed that RADB contains a rule with `REFINE` that merges an action `a` and an action `b` of the same `rp-attribute` in the "cartasian product", such that one of `a` and `b` is a method call and the other is an assigned set.

This makes little sense to me. I only noticed it because it caused the lexer to produce invalid AST that caused the parser to return early with an error.

**Comment by cunha** [comment]
Can you please paste the rule here so we can have a look?

**Reply by SichangHe** [reply]
Unfortunately no. I inferred its existence without seeing it.

**Reply by cunha** [reply]
Hard for me to say if it makes sense or not. I think we should have some tests to check the code. The logic is very complex, and without tests it may not be doing exactly what we'd want it to do. Looking at these weird cases and how the code handles is useful too as one-of validation checks.

**Reply by SichangHe** [reply]
If we were to do it, we would use the list of rules with `REFINE` in https://github.com/SichangHe/internet_route_verification/issues/104#issuecomment-1895296265.

**Comment by cunha** [comment]
We can use true cases, but it would likely be easier to generate synthetic ones. The thing is that we would need to generate the rule `e.g., mp-import` and then example announcements (prefix and AS-paths) to validate the semantics. It's easier to create the announcements if the rule is simpler to reason about.
